src/viam/sdk/rpc/dial.hpp

```diff
@@ -41,15 +41,20 @@ class Credentials {
 class DialOptions {
    public:
     DialOptions()
-        : auth_entity_(boost::none), credentials_(boost::none), allow_insecure_downgrade_(false) {}
+        : auth_entity_(boost::none),
```

Comment:
This constructor could be improved. The initializations of `auth_entity_` and `credentials_` with `boost::none` are unnecessary, since that is the state of a default initialized `boost::optional`. And the initializers of `allow_insecure_downgrade_` and `timeout_` would be better done as non-static data member initializers below, which would have the added benefit of moving the text explaining the choice of `20.0` next to its application:

```
...
bool allow_insecure_downgrade_ = false;
...
float timeout_ = 20.0f;
```

That'd make it possible to define the constructor as `DialOptions::DialOptions() = default` in `dial.cpp` which would also de-inline the constructor, which is probably a good idea.